tutorials/15-gaussian-processes/15_gaussian_processes.jmd

```diff
@@ -0,0 +1,161 @@
+---
+title: Introduction to Gaussian Processes using JuliaGPs and Turing.jl
+permalink: /tutorials/:name/
+redirect_from: tutorials/15-gaussian-processes/
+weave_options:
+  error : false
+---
+
+[JuliaGPs](https://github.com/JuliaGaussianProcesses/#welcome-to-juliagps) packages integrate well with Turing.jl because they implement the Distributions.jl
+interface.
+You should be able to understand what is going on in this tutorial if you know what a GP is.
+For a more in-depth understanding of the
+[JuliaGPs](https://github.com/JuliaGaussianProcesses/#welcome-to-juliagps) functionality
+used here, please consult the
+[JuliaGPs](https://github.com/JuliaGaussianProcesses/#welcome-to-juliagps) docs.
+
+In this tutorial, we will model the putting dataset discussed in chapter 21 of
+[Bayesian Data Analysis](http://www.stat.columbia.edu/%7Egelman/book/).
+The dataset comprises the result of measuring how often a golfer successfully gets the ball
+in the hole, depending on how far away from it they are.
+The goal of inference is to estimate the probability of any given shot being successful at a
+given distance.
+
+Let's download the data and take a look at it:
+
+```julia
+using CSV, DataDeps, DataFrames
+
+ENV["DATADEPS_ALWAYS_ACCEPT"] = true
+register(
+    DataDep(
+        "putting",
+        "Putting data from BDA",
+        "http://www.stat.columbia.edu/~gelman/book/data/golf.dat",
+        "fc28d83896af7094d765789714524d5a389532279b64902866574079c1a977cc",
+    ),
+)
+
+fname = joinpath(datadep"putting", "golf.dat")
+df = CSV.read(fname, DataFrame; delim=' ', ignorerepeated=true)
+df[1:5, :]
+```
+
+We've printed the first 5 rows of the dataset (which comprises only 19 rows in total).
+Observe it has three columns:
+
+ 1. `distance` -- how far away from the hole. I'll refer to `distance` as `d` throughout the rest of this tutorial
+ 2. `n` -- how many shots were taken from a given distance
+ 3. `y` -- how many shots were successful from a given distance
+
+We will use a Binomial model for the data, whose success probability is parametrised by a
+transformation of a GP. Something along the lines of:
+$$
+f \sim \operatorname{GP}(0, k) \\
+y_j \mid f(d_j) \sim \operatorname{Binomial}(n_j, g(f(d_j))) \\
+g(x) := \frac{1}{1 + e^{-x}}
+$$
+
+To do this, let's define our Turing.jl model:
+
+```julia
+using AbstractGPs, LogExpFunctions, ReverseDiff, Turing
+
+Turing.setadbackend(:reversediff) # use ReverseDiff for AD
+Turing.setrdcache(true)           # cache the tape between runs
```

Comment:
Maybe move this to the block for the `sample` call and make a brief mention that "We'll be using `ReverseDiff` here due to better performance on this example than the default `ForwardDiff`. See [insert link to docs] for more on performance in Turing.jl." or someting?

Reply:
Great idea. Will do now